### PR TITLE
HDF5 engine

### PR DESCRIPTION
**Pull request by ashishpriyadarshiCIC**
Co-authored-by: Harshit Bansal <harshit.bansalec@gmail.com>

### CODE REVIEW

**Comment by ashishpriyadarshiCIC** [comment]
Sample run :- 
```j
usage: retriever install [-h] [--compile] [--debug] [--not-cached]
                         {mysql,postgres,sqlite,msaccess,csv,json,xml,hdf5}
                         ...

positional arguments:
  {mysql,postgres,sqlite,msaccess,csv,json,xml,hdf5}
                        engine-specific help
    mysql               MySQL
    postgres            PostgreSQL
    sqlite              SQLite
    msaccess            Microsoft Access
    csv                 CSV
    json                JSON
    xml                 XML
    hdf5                HDF5

optional arguments:
  -h, --help            show this help message and exit
  --compile             force re-compile of script before downloading
  --debug               run in debug mode
  --not-cached          overwrites local cache of raw data
```

Example :- `retriever install hdf5 iris`

File: retriever/__main__.py [hunk 1]
```diff
@@ -238,6 +238,18 @@ def main():
             for dataset in scripts:
                 print("=> Installing", dataset.name)
                 try:
+                    if engine.name == "HDF5":
```
Comment (henrykironde) [inline retriever/__main__.py:241]: Note: Move this to the interface

Comment (henrykironde) [reply retriever/__main__.py:241]: I feel like we can remove this whole  block and use the functions that are provided by the interface 

```python
        if scripts:
            if args.dataset.endswith('.zip') or (hasattr(args, 'hash_value') and
                                                 args.hash_value):
                _install(vars(args), debug=debug, use_cache=use_cache)
                return
            for dataset in scripts:
                print("=> Installing", dataset.name)
                try:
                    if engine.name == "HDF5":
                        sqlite_opts = {
                            'command': 'install',
                            'dataset': dataset,
                            'engine': 'sqlite',
                            'file': (args.file.split("."))[0] + ".db",
                            'table_name': args.table_name,
                            'data_dir': args.data_dir
                        }
                        sqlite_engine = choose_engine(sqlite_opts)
                        dataset.download(sqlite_engine, debug=debug)
                        dataset.engine.final_cleanup()
                    dataset.download(engine, debug=debug)
                    dataset.engine.final_cleanup()
                except KeyboardInterrupt:
                    pass
                except Exception as e:
                    print(e)
                    if debug:
                        raise
            print("Done!")

```

to something like 

```python
        if scripts:
            _install(vars(args), debug=debug, use_cache=use_cache)
```
what do you think @apoorvaeternity

File: retriever/engines/hdf5.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        ("file", "Enter the filename of your HDF5 file", "hdf5.h5"),
+        ("table_name", "Format of table name", "{db}_{table}"),
+        ("data_dir", "Install directory", DATA_DIR),
+    ]
+
+    def create_db(self):
+        """Override create_db since an SQLite dataset needs to be created
+        first followed by the creation of an empty HDFStore file.
+        """
+        from retriever.engines.sqlite import engine
```
Comment (henrykironde) [inline retriever/engines/hdf5.py:28]: @ashishpriyadarshiCIC we do not need this import

File: test/test_regression.py [hunk 3]
```diff
@@ -35,7 +35,7 @@
     mysqldb_host = "mysqldb_retriever"
 
 mysql_engine, postgres_engine, sqlite_engine, msaccess_engine, \
-csv_engine, download_engine, json_engine, xml_engine = engine_list
+csv_engine, download_engine, json_engine, xml_engine, _ = engine_list
```
Comment (henrykironde) [inline test/test_regression.py:38]: why are we adding `_ `

File: test/test_provenance.py [hunk 4]
```diff
@@ -11,7 +11,7 @@
 
 file_location = os.path.normpath(os.path.dirname(os.path.realpath(__file__)))
 mysql_engine, postgres_engine, sqlite_engine, msaccess_engine, \
-csv_engine, download_engine, json_engine, xml_engine = engine_list
+csv_engine, download_engine, json_engine, xml_engine, _ = engine_list
```
Comment (henrykironde) [inline test/test_provenance.py:14]: why are we adding _

**Comment by henrykironde** [comment]
Thank you @harshitbansal05 and @ashishpriyadarshiCIC for working on this

**Pull request merged**